app/src/main/java/com/philkes/notallyx/utils/security/LockUtils.kt

```diff
@@ -69,19 +69,74 @@ private fun showBiometricOrPinPrompt(
     onFailure: () -> Unit,
 ) {
     when {
-        Build.VERSION.SDK_INT >= Build.VERSION_CODES.P -> {
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.R -> {
+            // Android 11+ with BiometricPrompt and Authenticators
+            val prompt = BiometricPrompt.Builder(context)
+                .apply {
+                    setTitle(context.getString(titleResId))
+                    descriptionResId?.let {
+                        setDescription(context.getString(descriptionResId))
+                    }
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
```

Comment:
The when case is equivalent to the if condition, so this if is always true, you can just remove the if condition and the else branch, since the else will never be called

Reply:
I originally put the if condition inside the when block because I wanted to make sure the code works well across different Android versions, especially Android 11+ (SDK 30+). I added the extra check to be clear about how `setAllowedAuthenticators` should behave on newer versions. But after thinking about it, I realize the when block already covers that check, so the extra condition was unnecessary. Thanks for pointing that out!

Reply:
No worries.

https://github.com/PhilKes/NotallyX/blob/74e5d051cf8b33a3d1a5ee2ad0eb0351dbd50cd2/app/src/main/java/com/philkes/notallyx/utils/security/LockUtils.kt#L84-L89

But you have to remove the `setNegativeButton` call here altogether, otherwise it throws following exception e.g. on my Android 14 emulator:

```java
java.lang.IllegalArgumentException: Can't have both negative button behavior and device credential enabled
at android.hardware.biometrics.BiometricPrompt$Builder.build(BiometricPrompt.java:485)
at com.philkes.notallyx.utils.security.LockUtilsKt.showBiometricOrPinPrompt(LockUtils.kt:91)
```